src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/scanner/StreamReader.kt

```diff
@@ -190,7 +191,7 @@ class StreamReader(
 
     private fun update() {
         try {
-            val buffer = stream.readUtf8()
+            val buffer = stream.readUtf8WithLimit(1024L)
```

Comment:
Let's discuss the value of `1024L`. I presume it's a trade-off: lower values would mean higher overhead in the internals of `readUtf8WithLimit`, and higher values would increase memory consumption. I'm wondering if increasing this value could lead to better performance. If you think alike, how about testing it on e.g. `1024L * 1024L`, and maybe some intermediate value?

Regardless of what value we'll figure out here, it's worth dropping a comment about this value.

Reply:
I was actually thinking about moving this into `LoadingSettings` (as it is in SnakeYAML).
But for sure, I can check how different sizes impact performance in our benchmarks

Reply:
I did a quick the with `1024 * 1024` buffer size - does not look like it makes the benchmark work any faster.
I will do additional checks tomorrow and let you know the results

Reply:
Thanks, for me, relying on snakeyaml-engine's default is good enough, and anything more would mean optimizing how snakeyaml-engine does it which is fine, but perhaps a topic for a separate PR.